### DISCUSSION TITLE
How to disable M-num bound to spacemacs/winum-select-window-num function in magit buffer

### DISCUSSION THREAD

**Discussion by c02y** [syl20bnr/spacemacs#15110] [Q&A]
This is not a bug report, so I'll skip the bug-report template.

FYI:
GNU Emacs 28.0.50
OS: Manjaro-Linux
Updating packages and spacemacs: DONE
Spacemcs version: commit 5c88c3b089c132

I managed to disable bindings like `M-1` ... `M-9` which switch windows using `winum` pacakge like:
```
  (dolist (M-num '("M-1" "M-2" "M-3" "M-4" "M-5" "M-6" "M-7" "M-8" "M-9"))
    (define-key winum-keymap (kbd M-num) nil))
```

But I cannot find a way to disable them in magit buffer, specifically in `magit-status-mode` buffer.
The only way I know is to comment the following part out in the spacemacs code base:
https://github.com/syl20bnr/spacemacs/blob/5c88c3b089c132997bf9f571a0207d93739efef1/layers/%2Bsource-control/git/packages.el#L265-L273

Originally bindings like `M-1/2/3/4` are bound to `magit-section-show-level-xxx` functions in `magit-status-mode` buffer, I want to keep them, and most importantly, I don't want `M-num` bindings bound to winum functions anywhere in Spacemacs, I'd like to just use `SPC num` simply.

Thank you in advance.

**Comment by arifer612** [comment]
Perhaps you could apply the same approach from before? Have the following in your `user-config` since it should only be loaded up after `magit` is configured. This way, you can avoid having to comment out whole blocks in the layer files.

```
(with-eval-after-load 'magit
  (dolist (M-num '("M-1" "M-2" "M-3" "M-4" "M-5" "M-6" "M-7" "M-8" "M-9"))
    (evil-define-key 'normal magit-section-mode-map (kbd M-num) nil)))
```

**Reply by c02y** [reply]
Thanks, it seems I already fixed this issue by using this long time ago, by keeping them for switching windows instead of disabling them.
```
(with-eval-after-load 'magit
    ;; overwrite the default M-1/2/3/4 since they are used for winum like in other modes
    ;; Use C-tab or S-tab to show section-cycle
    (bind-keys :map magit-mode-map
               ("M-1" . winum-select-window-1)
               ("M-2" . winum-select-window-2)
               ("M-3" . winum-select-window-3)
               ("M-4" . winum-select-window-4)))
```

**Comment by pataquets** [comment]
Looking for this too. I wanted to bind `M-7` and `M-8` to `(sp-backward-sexp)` and `(sp-forward-sexp)` in my Spacemacs `user-config`, but they keep doing `winum-select-...` unless I turn `winum-mode` off.
